pkg/restore/restore.go

```diff
@@ -713,23 +700,22 @@ func getNamespace(logger logrus.FieldLogger, path, remappedName string) *v1.Name
 	}
 }
 
-// TODO: this should be combined with DeleteItemActions at some point.
-func (ctx *restoreContext) getApplicableActions(groupResource schema.GroupResource, namespace string) []resolvedAction {
-	var actions []resolvedAction
+func (ctx *restoreContext) getApplicableActions(groupResource schema.GroupResource, namespace string) []framework.RestoreItemResolvedAction {
+	var actions []framework.RestoreItemResolvedAction
 	for _, action := range ctx.actions {
-		if !action.resourceIncludesExcludes.ShouldInclude(groupResource.String()) {
-			continue
-		}
-
-		if namespace != "" && !action.namespaceIncludesExcludes.ShouldInclude(namespace) {
-			continue
+		if action.ShouldUse(groupResource, namespace, nil, ctx.log) {
+			actions = append(actions, action)
 		}
+	}
+	return actions
+}
 
-		if namespace == "" && !action.namespaceIncludesExcludes.IncludeEverything() {
-			continue
+func (ctx *restoreContext) getApplicableItemSnapshotters(groupResource schema.GroupResource, namespace string) []framework.ItemSnapshotterResolvedAction {
```

Comment:
Is it func used anywhere?  Plus, how about merge this into `getApplicableActions` and make it return different types of resolvedActions?

Reply:
This is the fun part because Go won't let you cast arrays.  I went around a number of times trying to get this to merge but it was pretty ugly.  Trying to write a function that would work on both arrays meant I needed to pass in an array of Applicable and return an array of Applicable.  I wanted to keep the two arrays typed differently so that we wouldn't get the plugins mixed up anywhere.  If you can come up with a way to do it I'm open, it would be nice to reduce the duplicated code.

Reply:
It's fine to keep them in separate funcs if we don't want to mix up the plugins.